src/serverenvironment.cpp

```diff
@@ -469,6 +481,11 @@ void ServerEnvironment::init()
 
 	m_player_database = openPlayerDatabase(player_backend_name, m_path_world, conf);
 	m_auth_database = openAuthDatabase(auth_backend_name, m_path_world, conf);
+
+	if (m_map && m_script->has_on_mapblocks_changed()) {
+		m_map->addEventReceiver(&m_on_mapblocks_changed_receiver);
+		m_on_mapblocks_changed_receiver.receiving = true;
+	}
```

Comment:
question is: is it okay if registering a new callback at runtime silently fails?
IIRC no other callbacks behave this way.

Reply:
It is different, but I don't want people to pay the price of accumulating changed mapblocks if no mod uses the information. If a modder encounters this quirk, they can work around it. I could document it.

Reply:
I have now added this to the documentation.

Reply:
Would it be too expensive to make this an error? "register_on_mapblocks changed must be called at load time"

Reply:
I have now done as you suggest (a little while ago actually.)